### PR TITLE
r/vapp_entity: Add vapp_entity resource

### PR DESCRIPTION
**Pull request by bill-rich**

### CODE REVIEW

File: website/docs/r/vapp_entity.html.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+layout: "vsphere"
+page_title: "VMware vSphere: vsphere_vapp_entity"
+sidebar_current: "docs-vsphere-resource-compute-vapp-entity"
+description: |-
+  Provides a vSphere vApp entity resource. This can be used to describe the behavior of an entity (virtual machine or sub-vApp container) in a vApp container.
+---
+
+# vsphere\_vapp\_entity
```
Comment (bflad) [inline website/docs/r/vapp_entity.html.markdown]: Nit: Forward slashes in documentation headers are no longer required

File: website/docs/r/vapp_entity.html.markdown [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+layout: "vsphere"
+page_title: "VMware vSphere: vsphere_vapp_entity"
+sidebar_current: "docs-vsphere-resource-compute-vapp-entity"
```
Comment (bflad) [inline website/docs/r/vapp_entity.html.markdown]: Does this need a sidebar link in `website/vsphere.erb`?

File: website/docs/r/vapp_entity.html.markdown [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  Default: powerOn
+* `start_delay` - (Optional) Delay in seconds before continuing with the next
+  entity in the order of entities to be started. Default: 120 
+* `stop_action` - (Optional) Defines the stop action for the entity. Can be set
+  to none, powerOff, guestShutdown, or suspend. If set to none, then the entity
+  does not participate in auto-stop. Default: powerOff 
+* `stop_delay` - (Optional) Delay in seconds before continuing with the next
+  entity in the order sequence. This is only used if the stopAction is
+  guestShutdown. Default: 120 
+* `wait_for_guest` - (Optional) Determines if the	VM should be marked as being
```
Comment (bflad) [inline website/docs/r/vapp_entity.html.markdown]: Nit: Looks like some tabs found their way into here

File: website/docs/r/vapp_entity.html.markdown [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+the ID of the vApp Entity.
+
+[docs-import]: https://www.terraform.io/docs/import/index.html
+
+```
+terraform import vsphere_vapp_entity.vapp_entity vm-123:res-456 
+```
+
+The above would import the vApp entity that governs the behavior of the virtual
+machine with a MOID of vm-123 in the vApp container with the MOID res-456.
```
Comment (bflad) [inline website/docs/r/vapp_entity.html.markdown]: Nit: Might be better to spell out managed object ID instead of its abbreviation

**Comment by bill-rich** [comment]
```
make testacc TEST=github.com/terraform-providers/terraform-provider-vsphere/vsphere TESTARGS="-count=1 -run=TestAccResourceVSphereVAppEntity"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test github.com/terraform-providers/terraform-provider-vsphere/vsphere -v -count=1 -run=TestAccResourceVSphereVAppEntity -timeout 240m
=== RUN   TestAccResourceVSphereVAppEntity_basic
--- PASS: TestAccResourceVSphereVAppEntity_basic (19.00s)
=== RUN   TestAccResourceVSphereVAppEntity_nonDefault
--- PASS: TestAccResourceVSphereVAppEntity_nonDefault (17.10s)
=== RUN   TestAccResourceVSphereVAppEntity_update
--- PASS: TestAccResourceVSphereVAppEntity_update (25.87s)
=== RUN   TestAccResourceVSphereVAppEntity_multi
--- PASS: TestAccResourceVSphereVAppEntity_multi (20.35s)
=== RUN   TestAccResourceVSphereVAppEntity_multiUpdate
--- PASS: TestAccResourceVSphereVAppEntity_multiUpdate (29.40s)
=== RUN   TestAccResourceVSphereVAppEntity_import
--- PASS: TestAccResourceVSphereVAppEntity_import (26.93s)
PASS
ok  	github.com/terraform-providers/terraform-provider-vsphere/vsphere	138.672s
```

**Pull request merged**